HW1/P2/P2a.py

```diff
@@ -1,3 +1,61 @@
-from P2 import *
+# Default partitioning: 100 partitions!
 
-# Your code here
+import matplotlib.pyplot as plt
+
+import P2
+import seaborn as sns
+sns.set_context('poster', font_scale=1.25)
+import findspark as fs
+fs.init()
+import pyspark as ps
+import multiprocessing as mp
+import numpy as np
+
+# Setup cluster, number of threads = 2x cores
+config = ps.SparkConf()
+config = config.setMaster('local[' + str(2*mp.cpu_count()) + ']')
+config = config.setAppName('P2a')
+
+sc = ps.SparkContext(conf=config)
+
+# Do the computation
+
+num_pixels = 2000
+rows = sc.range(num_pixels, numSlices=10)
+cols = sc.range(num_pixels, numSlices=10)
+
+indices = rows.cartesian(cols)
```

Comment:
Since you use cartesian and rows and columns are identical, you really only need to use sc.range() once to generate rows and then do:

indices = rows.cartesian(rows)